### PR TITLE
Improve Auth error message for archived repos

### PR DESCRIPTION
**Pull request by agisilaos**
Fixes: #3077

As we discussed with @iAmWillShepherd we thought of this [improved] error message;

``Authentication failed. You may not have permission to access the repository or the repository has been archived. Open preferences and verify that you're signed in with an account that has permission to access this repository.`` 

Let us know what you think and if we can improve the message even more.

### CODE REVIEW

File: app/src/lib/git/core.ts [hunk 1]
```diff
@@ -184,7 +184,7 @@ function getDescriptionForError(error: DugiteError): string {
     case DugiteError.SSHAuthenticationFailed:
     case DugiteError.SSHPermissionDenied:
     case DugiteError.HTTPSAuthenticationFailed:
-      return `Authentication failed. You may not have permission to access the repository. Open ${__DARWIN__
+      return `Authentication failed. You may not have permission to access the repository or the repository may has been archived. Open ${__DARWIN__
```
Comment (shiftkey) [inline app/src/lib/git/core.ts]: Typo: 

> or the repository may **have** been archived

Comment (agisilaos) [reply app/src/lib/git/core.ts]: 👍

**Comment by agisilaos** [comment]
@shiftkey I think it's ready to be merged 🌮

**Comment by shiftkey** [comment]
@agisilaos this looks good to me, but I had a chat with @niik about this and how the message is rather complex already - I'll give him a chance to see if there's anything we can do better here before we merge it in.

**Comment by agisilaos** [comment]
Sounds like a plan. Yeah, let's do that.

**Comment by agisilaos** [comment]
Hey @niik, got any updates on this PR? Had you some time to review it etc? If yes, it's ok or I need to make any other changes to the error message?

**Comment by CaitinChen** [comment]
I meet the same problem. "Authentication failed. You may not have permission to access the repository or the repository may have been archived. Open preferences and verify that you're signed in with an account that has permission to access this repository."  
Please help me!! Thanks!!!!!!!!!!!!!!!!!! @agisilaos  @shiftkey

**Comment by shiftkey** [comment]
@CaitinChen please open a new issue and fill out the template

**Pull request merged**